src/query.rs

```diff
@@ -195,6 +195,12 @@ impl Query {
         self.client.add_option(name, value);
         self
     }
+
+    pub fn with_param(self, name: &str, value: impl Serialize) -> Result<Self, String> {
```

Comment:
This method must be well documented. Also, we should specify the difference between `param()` and `bind()` (probably in a dedicated section on `Query` and provide links to that section in docs of both methods).

Reply:
added some docs, signature should somewhat communicate difference given ? is positional while server side parameters are key/value

Reply:
Please, wrap into backticks `{name: type}`